subprojects/packagefiles/libupnp/meson.build

```diff
@@ -0,0 +1,292 @@
+auto.set('UPNP_VERSION_PATCH', ver[2])
+
+if get_option('blocking_tcp_connectionsr')
+  auto.set('UPNP_ENABLE_BLOCKING_TCP_CONNECTIONS', 1)
+endif
+
+if get_option('client')
+  auto.set('UPNP_HAVE_CLIENT', 1)
+endif
+
+if get_option('device')
+  auto.set('UPNP_HAVE_DEVICE', 1)
+endif
+
+if get_option('gena')
+  auto.set('UPNP_HAVE_GENA', 1)
+endif
+
+if get_option('tools')
+  auto.set('UPNP_HAVE_TOOLS', 1)
+endif
+
+if get_option('ipv6')
+  auto.set('UPNP_ENABLE_IPV6', 1)
+endif
+
+if get_option('optssdp')
+  auto.set('UPNP_HAVE_OPTSSDP', 1)
+endif
+
+if get_option('open_ssl')
+  auto.set('UPNP_ENABLE_OPEN_SSL', 1)
+endif
+
+if get_option('soap')
+  auto.set('UPNP_HAVE_SOAP', 1)
+endif
+
+if get_option('ssdp')
+  auto.set('UPNP_HAVE_SSDP', 1)
+endif
+
+if get_option('unspecified_server')
+  auto.set('UPNP_ENABLE_UNSPECIFIED_SERVER', 1)
+endif
+
+if get_option('webserver')
+  auto.set('UPNP_HAVE_WEBSERVER', 1)
+endif
+
+if get_option('reuseaddr')
+  auto.set('UPNP_MINISERVER_REUSEADDR', 1)
+endif
+
+if get_option('debug')
+  auto.set('UPNP_HAVE_DEBUG', 1)
+endif
+
+cc = meson.get_compiler('c')
+
+if cc.has_header_symbol('pthread.h', 'pthread_rwlock_t', prefix : '#define _XOPEN_SOURCE 500')
+  auto.set('UPNP_USE_RWLOCK', 1)
+endif
+
+if cc.has_header_symbol('string.h', 'strndup')
+  auto.set('HAVE_STRNDUP', 1)
+endif
+
+if cc.has_header_symbol('string.h', 'strnlen')
+  auto.set('HAVE_STRNLEN', 1)
+endif
+
+ufile = configure_file(
+  output : 'upnpconfig.h',
+  configuration : auto
+)
+
+cfile = configure_file(
+  output : 'autoconfig.h',
+  configuration : auto
+)
+
+deps = []
+if host_machine.system() == 'windows'
+  deps += cc.find_library('ws2_32')
+  deps += cc.find_library('iphlpapi')
+elif host_machine.system() == 'sunos'
+  deps += cpp.find_library('nsl')
+  deps += cpp.find_library('socket')
+endif
+deps += dependency('libcrypto', required : get_option('open_ssl'))
+deps += dependency('libssl', required : get_option('open_ssl'))
+threads_dep = dependency('threads')
+
+libupnp = library('upnp',
+  upnp_sources,
+  include_directories : [ libixml_incdir, libupnp_incdir ],
+  dependencies : [ deps, threads_dep ],
+  link_with : libixml
+)
+
+pkg = import('pkgconfig')
+pkg.generate(libupnp)
+
+headers = [
+  'ixml/inc/ixml.h',
+  'ixml/inc/ixmldebug.h',
+  'upnp/inc/Callback.h',
+  'upnp/inc/list.h',
+  'upnp/inc/upnp.h',
+  'upnp/inc/UpnpActionComplete.h',
+  'upnp/inc/UpnpActionRequest.h',
+  'upnp/inc/UpnpDiscovery.h',
+  'upnp/inc/UpnpEvent.h',
+  'upnp/inc/UpnpEventSubscribe.h',
+  'upnp/inc/UpnpExtraHeaders.h',
+  'upnp/inc/UpnpFileInfo.h',
+  'upnp/inc/UpnpGlobal.h',
+  'upnp/inc/UpnpInet.h',
+  'upnp/inc/UpnpIntTypes.h',
+  'upnp/inc/UpnpStateVarComplete.h',
+  'upnp/inc/UpnpStateVarRequest.h',
+  'upnp/inc/UpnpStdInt.h',
+  'upnp/inc/UpnpString.h',
+  'upnp/inc/UpnpSubscriptionRequest.h',
+  'upnp/inc/UpnpUniStd.h',
+  ufile,
+]
+
+if get_option('tools')
+  'upnp/inc/upnptools.h'
+endif
+
+install_headers(headers)
```

Comment:
It seems that upstream installs all these to a 'upnp' subdirectory of the includedir?

Reply:
I'll need to check this.